presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PropertyDerivations.java

```diff
@@ -123,8 +126,15 @@ public static ActualProperties deriveProperties(PlanNode node, List<ActualProper
         verify(node.getOutputSymbols().containsAll(localPropertyColumns), "Node-level local properties contain columns not present in node's output");
 
         // TODO: ideally this logic would be somehow moved to PlanSanityChecker
-        verify(node instanceof SemiJoinNode || inputProperties.stream().noneMatch(ActualProperties::isNullsReplicated) || output.isNullsReplicated(),
-                "SemiJoinNode is the only node that can strip null replication");
+        if (!(node instanceof SemiJoinNode)) {
+            Set<Replication> inputReplications = inputProperties.stream()
+                    .map(ActualProperties::getReplication)
+                    .filter(replication -> !replication.isNoReplication())
+                    .collect(toImmutableSet());
+            verify(inputReplications.size() == 0
+                            || (inputReplications.size() == 1 && Objects.equals(output.getReplication(), getOnlyElement(inputReplications))),
```

Comment:
I don't exactly know context of it, but it seems that either:
* none input should be replicated
* the only one input should be replicated the same as output

Here you have:
some replicates inputs should be replicated the same as output. That is: there might be unreplicated input left.